src/Prettus/Repository/Eloquent/BaseRepository.php

```diff
@@ -450,20 +450,19 @@ public function firstOrCreate(array $attributes = [])
     }
 
     /**
-     * Set the "limit" value of the query.
+     * Retrieve data of repository with limit applied
+     *
+     * @param int $limit
+     * @param array $columns
      *
-     * @param  int  $value
      * @return mixed
      */
-    public function limit($limit)
+    public function limit($limit, $columns = ['*'])
     {
-        $this->applyCriteria();
-        $this->applyScope();
-        $results = $this->model->limit($limit);
-
-        $this->resetModel();
+        // Shortcut to all with `limit` applied on query via `take`
+        $this->take($limit);
 
-        return $this->parserResult($results);
+        return $this->all($columns);
```

Comment:
:boom: Breaking change, in one of our project we are doing `$repository->limit(10)->get()`.

After updating to `2.7.6`, it now throws `Too few arguments to function Illuminate\Support\Collection::get(), 0 passed in /home/deploy/www/vivamusicgroup/app/Http/Composers/Backend/DashboardComposer.php on line 85 and at least 1 expected`.

Reply:
Downgrading to 2.7.5 what object is returned when using `$repository->limit(10)`?

I don't get how `parserResult()` is returning a different object there.

Reply:
> Downgrading to 2.7.5 what object is returned when using $repository->limit(10)

It returns a `Illuminate/Database/Eloquent/Builder`.

> I don't get how `parserResult()` is returning a different object there.

I'm not quite sure either, I haven't used and `Presenters` in my projects yet...

https://github.com/andersao/l5-repository/blob/ad55bf207d96abe23fcdb5cc7156d06b1984cf10/src/Prettus/Repository/Eloquent/BaseRepository.php#L1074-L1095

Reply:
Oh, i see... without presenters parseResult parses nothing, and only return the builder. The old limit method causes this mislead.

I don’t know how can this be fixed, for me this is a bug on parseResult. @bsormagec  can you help us?

@juliomotol Is easy for you adapt to use ‘all’ where ‘get’ is used with limit?
> $repository->limit(10)->all()

Reply:
The breaking change was that the query was concluded when the `all()` method was called.

Prior to this commit, we still needed to chain the `all()`/`get()` after `limit()`.

We've already updated all the repositories to no longer chain `all()`/`get()` after `limit()`.

These comments are just to notify you that this is a breaking change in our case when we upgraded to the latest version.

@andersao @bsormagec